lib/fitbank_api/pix/dynamic_qr_code.rb

```diff
@@ -0,0 +1,184 @@
+          base_url: String,
+          receiver_bank_info: FitBankApi::Entities::BankInfo,
+          credentials: FitBankApi::Entities::Credentials,
+          receiver_pix_key: String,
+          receiver_zip_code: String
+        ).void
+      end
+      # @param [String] base_url The base URL of the API, defining whether prod
+      #   or sandbox environemt is used
+      # @param [FitBankApi::Entities::BankInfo] receiver_bank_info Bank information
+      #   for the one receiving the money. In our case this is Latam's bank info
+      # @param [FitBankApi::Entities::Credentials] credentials Credentials used to
+      #   access the API
+      # @param [String] pix_key The PIX key of the one receiving the money.
+      # @param [String] receiver_zip_code Zip Code of the city where the receiver is situtated.
+      #   It is allowed to have a dash in the zip code.
+      # @note For sandbox the pix_key must be registerd into FitBank's database via:
+      #   https://dev.fitbank.com.br/reference/217. For productino we can use already existing
+      #   PIX key.
+      def initialize(
+        base_url:,
+        receiver_bank_info:,
+        credentials:,
+        receiver_pix_key:,
+        receiver_zip_code:
+      )
+        @generate_code_url = T.let(
+          URI.join(base_url, 'main/execute/GenerateDynamicPixQRCode'), URI::Generic
+        )
+        @get_code_by_id_url = T.let(
+          URI.join(base_url, 'main/execute/GetPixQRCodeById'), URI::Generic
+        )
+        @receiver_bank_info = receiver_bank_info
+        @credentials = credentials
+        @receiver_pix_key = receiver_pix_key
+        @receiver_zip_code = receiver_zip_code
+      end
+
+      sig do
+        params(
+          value: BigDecimal,
+          expiartion_date: Date,
+          id: String
+        ).returns(T::Hash[Symbol, T.untyped])
+      end
+      # Generate a dynamic QR Code which when scanned will transfer money
+      # to the receiver_bank_info.
+      # @param [BigDecimal] value The amount of the sum transfered when the QR code is scanned
+      # @param [Date] expiartion_date The QR code will not be valid after this date
+      # @param [String] id Custom identifier for the QR code. You cannot serch by it, nor
+      #   use it for retries. You can, however, see it when you query the API for the QR code info.
+      def generate(
+        value:,
+        expiartion_date:,
+        id:
+      )
+        # Endpoint documentation: https://dev.fitbank.com.br/reference/256
+        # Fields which are not required are ommited for now
+        # TODO: Check if passing receiver name and cpf will force FitBank to do the name/cpf checked
+        payload = {
+          Method: 'GenerateDynamicPixQRCode',
+          PartnerId: @credentials.partner_id,
+          BusinessUnitId: @credentials.business_unit_id,
+          PixKey: @receiver_pix_key,
+          TaxNumber: @credentials.cnpj,
+          PrincipalValue: value.truncate(2).to_s('F'),
```

Comment:
I think the value truncation or rounding is the responsibility of the caller.

Reply:
It's working with big decimal which has exponential string representation. This crashes the API, so we have to format it in a suitable form.

Reply:
I'm talking about the call to `truncate`, not the `to_s` call.